sdk/include/opentelemetry/sdk/http/http_client.h

```diff
@@ -0,0 +1,153 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+#include "opentelemetry/http/http_client.h"
+#include "opentelemetry/version.h"
+
+#include <algorithm>
+#include <cstring>
+#include <functional>
+#include <map>
+#include <memory>
+#include <vector>
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace http
+{
+namespace http_api = opentelemetry::http;
+
+static std::function<bool(const std::string &, const std::string &)> CaseInsensitiveComparator =
+    [](const std::string &s1, const std::string &s2) -> bool {
+  std::string str1(s1.length(), ' ');
+  std::string str2(s2.length(), ' ');
+  auto lowerCase = [](char c) -> char { return tolower(c); };
+  std::transform(s1.begin(), s1.end(), str1.begin(), lowerCase);
+  std::transform(s2.begin(), s2.end(), str2.begin(), lowerCase);
+  return str1 < str2;
+};
+
+// HttpHeaders implementation
+class HttpHeaders : http_api::HttpHeaders,
+                    std::multimap<std::string, std::string, decltype(CaseInsensitiveComparator)>
```

Comment:
Should this `std::multimap` based on `nostd::string_view` according to `insert()` called in `add()` below?

Reply:
I'd actually advocate again that we can just use `std::string` here in SDK for the sake of simplicity and compat with older compilers, just using plain standard C++11 (or maybe C++14) classes, so that external components do not have to take a dependency on `nostd`. There is no ABI stability guarantee for exporters. We discussed it in prior SIG meetings that we are not providing ABI stable interface in SDK, for exporters, and thus, using `nostd` here is a bit of an overkill.

In short: there is no requirement to use `nostd` inside the SDK, because there is no ABI-stability requirement within the SDK.

Reply:
We definitely shouldn't store `nostd::string_view` in the `multimap`.

There are some slight advantages of using `nostd::string_view` over `std::string` as type for parameters in function calls, as this allows users to pass both `std::string` and `const char*` without creating an intermediate `std::string`. But I don't have a strong preference for it in this API, which will not be on a performance critical path.

Reply:
This would be changed in new PR. `string_view` class won't be used anymore. Although agree on `string_view` advantage as stated by @pyohannes